File: lab spring25/assignment4/taskD.cpp
#include <stdio.h>
#define rept(i,n) for(int i=0;i<n;i++)

void solve(){
    int n,m,odd=0;
    scanf("%d %d",&n,&m);
    int a,deg[n+1]={};
    rept(i,m*2) {
        scanf("%d",&a);
        deg[a]++;
        deg[a]&1?odd++:odd--;
    }
    printf((odd!=2 && odd!=0)?"NO\n":"YES\n");
}

int main(){
    freopen("input.txt","r",stdin);
    int t{1};
    scanf("%d",&t);
    while (t--) solve();
    return 0;
}

File: lab summer25/assignment4/taskD.py
import sys
sys.stdin = open("input.txt","r")

def solve():
    n,m = list(map(int,input().split()))
    u = list(map(int,input().split()))
    v = list(map(int,input().split()))
    deg = [0]*(n+1)
    for i in range(m):
        deg[u[i]]+=1
        deg[v[i]]+=1
    odd = 0
    for i in range(1,n+1): odd+=(deg[i]&1)
    print("YES") if (odd in (0,2)) else print("NO")

t = 1
t = int(input())
while t>0:
    t-=1
    solve()